### PR TITLE
feat: WYSIWYG markdown field editor

### LINKED ISSUE #1691: WYSIWYG Proposal Editor

### PR DESCRIPTION
**Pull request by 1emu**
Closes #1691 

image

image

### CODE REVIEW

File: src/components/Common/Form/MarkdownField.tsx [hunk 1]
```diff
@@ -1,30 +1,145 @@
...
+          aria-label="Preview"
+          onClick={click}
+        >
+          {'Preview'}
+        </Button>
+      )
+    }
+    return (
+      <Button type="button" className="MarkdownField__Command" data-name="edit" aria-label="Edit" onClick={click}>
+        {'Edit'}
```
Comment (ncomerci) [inline src/components/Common/Form/MarkdownField.tsx:60]: Shouldn't this be in en.json?

**Review by ncomerci — APPROVED** [review]

File: src/components/Common/Form/MarkdownField.tsx [hunk 2]
```diff
@@ -1,30 +1,145 @@
...
+    if (preview === 'edit') {
+      return (
+        <Button
+          type="button"
+          className="MarkdownField__Command"
+          data-name="preview"
+          aria-label="Preview"
+          onClick={click}
+        >
+          {'Preview'}
```
Comment (andyesp) [inline src/components/Common/Form/MarkdownField.tsx:54]: This one too to the i18n file.

**Review by andyesp — APPROVED** [review]

File: src/components/Common/Form/MarkdownField.tsx [hunk 3]
```diff
@@ -1,30 +1,145 @@
...
+              {...markdownProps}
+              preview={disabled ? 'preview' : 'edit'}
+              previewOptions={{
+                rehypePlugins: [[rehypeSanitize]],
+              }}
+              textareaProps={{ disabled: disabled }}
+              components={{
+                preview: (source: string) => (
+                  <div style={{ minHeight: '100%' }}>
+                    <Markdown>{source}</Markdown>
```
Comment (andyesp) [inline src/components/Common/Form/MarkdownField.tsx:127]: Can't you use ProposalMarkdown here?

**Review by andyesp — APPROVED** [review]
We should review the focus part.

**Pull request merged**